pkg/snowflake/tag_association.go

```diff
@@ -103,7 +92,7 @@ func (tb *TagAssociationBuilder) Drop() string {
 }
 
 func ListTagAssociations(tb *TagAssociationBuilder, db *sql.DB) ([]tagAssociation, error) {
-	stmt := fmt.Sprintf(`SELECT * FROM SNOWFLAKE.ACCOUNT_USAGE.TAG_REFERENCES WHERE TAG_NAME = '%v' AND DOMAIN = '%v' AND TAG_VALUE = '%v'`, tb.tagName, tb.objectType, tb.tagValue)
+	stmt := fmt.Sprintf(`SELECT SYSTEM$GET_TAG('"%v"."%v"."%v"', '%v', '%v') TAG_VALUE WHERE TAG_VALUE IS NOT NULL`, tb.databaseName, tb.schemaName, tb.tagName, tb.objectName, tb.objectType)
```

Comment:
I tried a minimal-invasive way to switch to the `SYSTEM$GET_TAG` function, probably it could be improved quite a bit (like checking if the returned `TAG_VALUE` is actually the same value as applied in the resource to detect diffs).

Behavior of the `SYSTEM$GET_TAG` function:
* If the tag identifier references a non-existing or unauthorized tag object, the query fails.
* If the object identifier references a non-existing or unauthorized object, the query fails.
* If the tag is not set on the object, the function returns a SQL NULL
* If the tag is set on the object, the function returns a string with the tag value.

I didn't expect using the column alias would work in the WHERE clause, but it does:
![image](https://user-images.githubusercontent.com/26149233/194546581-737e3245-956c-4e33-8f85-dedbbac64913.png)